lexicon/providers/yandexcloud.py

```diff
@@ -0,0 +1,307 @@
+        """
+        payload: Dict = self._get("/zones", {"folderId": folder_id, "filter": f"zone='{self.domain}.'"})
+
+        if "dnsZones" not in payload:
+            raise AuthenticationError(f"No domain found for Folder ID {folder_id}")
+        if len(payload["dnsZones"]) > 1:
+            raise AuthenticationError("Too many domains found, this should not happen")
+
+        return payload["dnsZones"][0]["id"]
+
+    def _list_records(self, rtype=None, name=None, content=None) -> List[Dict]:
+        """List all records. Return an empty list if no records found.
+        type, name and content are used to filter records.
+        https://cloud.yandex.com/en/docs/dns/api-ref/DnsZone/listRecordSets
+        """
+        records: List[Dict] = []
+
+        query_params = {"filter": ""}
+        if rtype:
+            query_params["filter"] = f"type='{rtype}'"
+
+        # name should be in full form, transform it in case it's not
+        if name:
+            name = self._full_name(name) + "."
+            query_params["filter"] += " AND " if query_params["filter"] else ""
+            query_params["filter"] += f"name='{name}'"
+
+        url = f"/zones/{self.domain_id}:listRecordSets"
+        more_pages = True
+        while more_pages:
+            payload: Dict = self._get(url, query_params)
+            if "nextPageToken" in payload:
+                query_params["pageToken"] = payload["nextPageToken"]
+            else:
+                more_pages = False
+
+            # nothing to list
+            if "recordSets" not in payload:
+                break
+
+            for record in payload["recordSets"]:
+                for entry in record["data"]:
+                    # artificially generate ID for deletion
+                    record_id = f"{record['type']}${self._relative_name(record['name'])}${entry}"
+                    records.append({
+                        "id": record_id,
+                        "name": self._full_name(record["name"]),
+                        "type": record["type"],
+                        "ttl": record["ttl"],
+                        "content": entry,
+                    })
+
+        if content:
+            records = [record for record in records if record["content"].lower() == content.lower()]
+
+        LOGGER.debug("list_records: %s", records)
+        return records
+
+    def _get_record(self, rtype: str, name: str) -> Dict:
+        """Gets a single record by rtype and name.
+        Throws HTTPError with status code 404 if record not found.
+        https://cloud.yandex.com/en/docs/dns/api-ref/DnsZone/getRecordSet"""
+
+        if name is not None:
+            name = self._full_name(name) + "."
+
+        url = f"/zones/{self.domain_id}:getRecordSet"
+        query_params = {"name": name, "type": rtype}
+        # name should be in full form, transform it in case it's not
+
+        payload: Dict = self._get(url, query_params)
+
+        return {
```

Comment:
We may return directly payload, it does not seem critical to leak extra params from this method.